giraffe/src/transforms/annotation.ts

```diff
@@ -37,25 +37,32 @@ export const annotationTransform = (
     }
   }
 
-  const annotationData: AnnotationMark[] = []
-
+  // A hash of all input annotations is needed to:
+  // - keep only the first occurence of an annotation with the same
+  //   dimension, startValue, stopValue
+  // - ensure 'y' dimension annotations are within yDomain
+  // - ensure 'x' dimension annotations are within xDomain
+  const annotationMap = {}
   if (Array.isArray(inputAnnotations)) {
     for (let i = 0; i < inputAnnotations.length; i += 1) {
-      // Keep only the annotations within their respective domains
-      // 'y' direction annotations should be within yDomain
-      // 'x' direction annotations should be within xDomain
       if (
-        (inputAnnotations[i].direction === 'y' &&
+        !annotationMap[
+          `${inputAnnotations[i].dimension}-${inputAnnotations[i].startValue}-${inputAnnotations[i].stopValue}`
```

Comment:
what's this big computed string all about?

Reply:
It's a unique key, to ensure we don't keep duplicate annotations that the user might have passed in. This can make a huge difference for performance when hovering.

I have refactored this string out to a descriptive variable.